catgrad/src/stdlib/nn/mod.rs

```diff
@@ -197,12 +197,17 @@ pub fn causal_mask(builder: &Builder, size: Var) -> Var {
     mask * ninf
 }
 
-pub fn linear_no_bias(builder: &Builder, in_dim: usize, out_dim: usize, p: Path, x: Var) -> Var {
-    let w = param(builder, &p.extend(["weight"]).unwrap());
-
+pub fn linear_b_param(
+    builder: &Builder,
+    in_dim: usize,
+    out_dim: usize,
+    weight: Var,
+    bias: Option<Var>,
+    x: Var,
+) -> Var {
     let dim0 = 0.to_nat(builder);
     let dim1 = 1.to_nat(builder);
-    let w_t = transpose(builder, dim0, dim1, w);
+    let w_t = transpose(builder, dim0, dim1, weight);
 
     let sh = shape(builder, x.clone());
     let [batch_size] = unpack::<1>(builder, sh);
```

Comment:
I think this is a bug- `unpack::<1>` should be giving an error if this is not a rank-1 shape. Need to fix the upstream bug too.